app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/MessageComposer.kt

```diff
@@ -53,21 +53,42 @@ import com.wire.android.ui.theme.wireColorScheme
 import com.wire.android.ui.theme.wireDimensions
 import com.wire.android.ui.theme.wireTypography
 
+
+/*
+* Message composer is a UI widget that handles the UI logic of sending of the messages,
+* it is a wrapper around the "hosting" widget. It takes in a messageText and
+* exposes a onMessageChanged lambda, giving us the option to control it's Message Text outside of the Widget.
+* it also exposes other lambda's giving us the option to handle the different message actions
+* */
 @OptIn(ExperimentalMaterialApi::class)
 @Composable
 fun MessageComposer(
     content: @Composable () -> Unit,
-    messageText: TextFieldValue,
-    onMessageChanged: (TextFieldValue) -> Unit,
+    messageText: String,
+    onMessageChanged: (String) -> Unit,
     onSendButtonClicked: () -> Unit
 ) {
-    val messageComposerState = rememberMessageComposerInnerState()
+    val messageComposerState = remember {
+        MessageComposerInnerState(
+            "",
+            MessageComposeInputState.Enabled
+        )
+    }
 
     MessageComposer(
         content = content,
         messageComposerState = messageComposerState,
-        messageText = messageText,
-        onMessageChanged = onMessageChanged,
+        // we are setting the TextFieldValue by copying the current value and updating the text
+        // in order to keep the correct state of cursor position etc ...
+        messageText = messageComposerState.textFieldValue.copy(text = messageText),
+        onMessageChanged = {
+            // we are setting it immediately in the UI first
+            messageComposerState.textFieldValue = it
+            // we are hoisting the TextFieldValue text value up to the parent
+            if (messageText != it.text) {
```

Comment:
I don't this logic is necessary, since the `BasicTextField` is doing it for us already

Reply:
should we then remove this line?

Reply:
we actually need this line otherwise it will nto be propagated to the parent